nixos/modules/programs/bat.nix

```diff
@@ -0,0 +1,59 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+let
+  inherit (lib.options) mkEnableOption mkPackageOption mkOption;
+  inherit (lib.modules) mkIf;
+  inherit (lib.lists) optionals;
+  inherit (lib.types) lines;
+  inherit (lib) maintainers;
+  cfg = config.programs.bat;
+in
+{
+  options.programs.bat = {
+    enable = mkEnableOption "`bat`, a {manpage}`cat(1)` clone with wings";
+    package = mkPackageOption pkgs "bat" { };
+    enableExtras = mkEnableOption "`bat`'s extra scripts and utilities";
+    # TODO: Somehow turn this into a structured submodule per RFC 0042.
```

Comment:
I still think a list would be better here

Reply:
My apologies, I'm not sure I understand the review. Do you mean turning the `extraConfig` into a list? (as opposed to an attrset) or the `enableExtras` option?

Reply:
Yes, extraConfig. It being lines feels strange because it is really just a list of arguments

Reply:
Not exactly sure if I did this correctly, but extraConfig is now a `listOf` `str`